elasticdl/pkg/ps/server.go

```diff
@@ -63,23 +71,53 @@ func createMasterClient(masterAddr string) *MasterClient {
 }
 
 // NewServer creates a Server instance
-func NewServer(ID int, optType string, optArgs string, masterAddr string, evaluationSteps int) *Server {
+func NewServer(ID int, optType string, optArgs string, masterAddr string,
+	evaluationStep int, checkpointDirForInit string,
+	checkpointDir string, checkpointStep int, keepCheckpointMax int, numPsPods int) *Server {
 	var ps Server
-	ps.Model = NewModel()
+	if checkpointDirForInit != "" {
+		var err error
+		ps.Model, err = LoadModelFromCheckpoint(checkpointDirForInit, ID, numPsPods)
+		ps.Model.Initialized = true
+		if err != nil {
+			log.Fatalf("failed to load from checkpoint: %v", err)
+		}
+	} else {
+		ps.Model = NewModel()
+	}
+
 	var err error
 	ps.Opt, err = NewOptimizer(optType, optArgs)
 	if err != nil {
 		log.Fatalf("failed to create PS server: %v", err)
 	}
 	ps.ID = ID
 	ps.masterClient = createMasterClient(masterAddr)
-	ps.evaluationSteps = int32(evaluationSteps)
+	ps.evaluationStep = evaluationStep
+	ps.checkpointDirForInit = checkpointDirForInit
+	ps.checkpointDir = checkpointDir
+	ps.checkpointStep = checkpointStep
```

Comment:
Duplicate with line 96

Reply:
line 96 is evaluationStep, this line is checkpointStep